internal/abi_test.go

```diff
@@ -0,0 +1,40 @@
+package internal
+
+import (
+	"github.com/cilium/ebpf"
+	"github.com/cloudflare/xdpcap"
+	"testing"
+)
+
+const hookSymbol = "foo"
+
+// Test CheckHookMap with valid and invalid HookMap ABI
+func TestCheckHookMap(t *testing.T) {
+	scpec := &ebpf.MapSpec{
+		Name:       ebpf.SanitizeName(hookSymbol, '_'),
+		Type:       xdpcap.HookMapABI.Type,
+		KeySize:    xdpcap.HookMapABI.KeySize,
+		ValueSize:  xdpcap.HookMapABI.ValueSize,
+		MaxEntries: 4,
+	}
+	valid, err := ebpf.NewMap(scpec)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer valid.Close()
+
+	if CheckHookMap(valid) != nil {
```

Comment:
This will report the wrong `error` in `t.Fatal()` I think, we need `if err := CheckHookMap(valid); err != nil {` to preserve the error returned by `CheckHookMap()`

Reply:
Oh! sorry my bad.
fixed